### PR TITLE
refactor: replace `H256` and `U256` with `UnsignedFieldElement` in VO's

### PR DESCRIPTION
**Pull request by xJonathanLEI**
Resolves task 2 of #13:

> replace `U256` and `H256` with `UnsignedFieldElement` in value objects and providers

This is a messy one, as we're fixing all the misuse of `H256` and `U256` across a lot of files.

### CODE REVIEW

File: starknet-core/src/types/transaction.rs [hunk 1]
```diff
@@ -26,8 +28,8 @@ pub struct TransactionWithStatus {
     pub transaction: Option<Transaction>,
     pub status: TransactionStatusType,
     #[serde(default)]
-    #[serde(deserialize_with = "deserialize_pending_block_hash")]
-    pub block_hash: Option<H256>,
+    #[serde(deserialize_with = "pending_block_hash_de")]
```
Comment (milancermak) [inline starknet-core/src/types/transaction.rs:31]: The `#[serde]`s here could be merge into one line.

**Review by milancermak — APPROVED** [review]
Looking good!

Kinda shame that the API uses both decimal and hex representations for the FieldElement values.

File: starknet-core/src/types/transaction.rs (hunk 1 above)
Comment (xJonathanLEI) [reply starknet-core/src/types/transaction.rs:31]: Yes ser

**Comment by xJonathanLEI** [comment]
Yeah should have just followed Ethereum and make everything hex.

**Pull request merged**